File: task1/src/services/storyService.js
import api from './api';

const storyService = {
    getAllStories: async () => {
        const response = await api.get('/stories');
        return response.data;
    },

    getStoryById: async (id) => {
        const response = await api.get(`/stories/${id}`);
        return response.data;
    },

    createStory: async (storyData) => {
        const response = await api.post('/stories', storyData);
        return response.data;
    },

    updateStory: async (id, storyData) => {
        const response = await api.put(`/stories/${id}`, storyData);
        return response.data;
    },

    deleteStory: async (id) => {
        const response = await api.delete(`/stories/${id}`);
        return response.data;
    }
};

export default storyService;

File: backend/src/main/java/com/storyboard/backend/service/StoryService.java
package com.storyboard.backend.service;

import com.storyboard.backend.entity.Story;
import java.util.List;

public interface StoryService {
    List<Story> getAllStories();

    Story getStoryById(Long id);

    Story createStory(Story story);

    Story updateStory(Long id, Story story);

    void deleteStory(Long id);
}
